### PR TITLE
Make im2col default option for quartus

### PR DESCRIPTION
**Pull request by calad0i**
A# Description

With using the Quartus backend, with conv2d with kernel size 3x3 or conv1d with kernel size 3, the framework makes the default implementation `Winograd` instad of `im2col`. As the current `Winograd` implementation does not seem to play nicely with low bitwidth and hard to achieve bit-accuracy (e.g., with automatic precision setting), I would suggest not enable it by default.

For the Winograd impl, type `uint8_t` from `<cstdint>` is used. This header is not included by default in many environments, explicit `#include <cstdint>` is added to corresponding files. 

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Documentation update
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] A new research paper code implementation
- [ ] Other (Specify)

## Checklist

- [ ] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by jmitrevs** [comment]
I don't have a strong opinion on this. Let's see what others think.

File: hls4ml/templates/quartus/firmware/nnet_utils/nnet_conv1d_resource.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include "nnet_common.h"
 #include "nnet_dense.h"
+#include <cstdint>
```
Comment (vloncar) [inline hls4ml/templates/quartus/firmware/nnet_utils/nnet_conv1d_resource.h:6]: What do we use from this header?

**Comment by vloncar** [comment]
I'm fine with this type of change. But last I remember, im2col often didn't even synthesize so we added some heuristic to at least try winograd if possible. @bo3z might remember better.

**Comment by bo3z** [comment]
> I'm fine with this type of change. But last I remember, im2col often didn't even synthesize so we added some heuristic to at least try winograd if possible. @bo3z might remember better.

Generally, some designs would fail HLS compilation and it just seemed that Winograd was more successful. This could be tied to a specific version of Intel HLS or some implementation detail we had at the time. If I recall correctly, the heuristic was such that if it's 3x3 (and no stride, padding etc.) we set it to use Winograd. But it's well known that Winograd doesn't perform well at extreme quantisations (which granted were never tested at the time).

File: hls4ml/templates/quartus/firmware/nnet_utils/nnet_conv1d_resource.h (hunk 1 above)
Comment (calad0i) [reply hls4ml/templates/quartus/firmware/nnet_utils/nnet_conv1d_resource.h:6]: This header provides the `uint8_t` used in that file, and is not included by default in all platforms. Though, I am not sure if using `uint8_t` provides any performance benefit as that index should probably be optimized out when compiling to hdl...

**Review by jmitrevs — APPROVED** [review]

**Pull request merged**